src/task2/task2.controller.ts

```diff
@@ -0,0 +1,52 @@
+import {
+    Controller,
+    Post,
+    Body,
+    Get,
+    Param,
+    Patch,
+    Delete,
+  } from '@nestjs/common';
+  
+  import { FlightService } from "./task2.services";
+
+@Controller('flight')
+export class FlightController {
+    constructor(private readonly flightService: FlightService) {}
+    @Post()
+    bookFlight(@Body('source') flightsource: string, 
+                @Body('destination') flightdestination: string, 
+                @Body('date') flightDate: string, 
+                @Body('fare') flightfare: number, 
+                @Body('seats') flightSeats: number, 
+                @Body('flightno') flightNo: string,
+                @Body('seatno') seatNo: number,
+                 ) {
+        this.flightService.bookFlight(flightsource, flightdestination,flightDate, flightfare, flightSeats, flightNo , seatNo);
+    }
+    @Delete()
+    cancel(@Body('customerid') id: string): string {
+        this.flightService.cancel(id); 
+        return "Flight has been cancelled";

```

Comment:
this about having response code and message too as your reponse object to client